external/loaders/loaders/batches/_batch.py

```diff
@@ -115,6 +120,11 @@ def batches_from_mapper(
         needs_grid: Add grid information into batched datasets. [Warning] requires
             remote GCS access
         in_memory: if True, load data eagerly and keep it in memory
+        unstacked_dims: if given, produce stacked and shuffled batches retaining
+            these dimensions as unstacked (non-sample) dimensions
+        drop_nans: if True, drop NaN values from the data, and raise an
+            exception if all values in a batch are NaN. requires unstacked_dims
+            argument is given, raises a ValueError otherwise.
```

Comment:
Since this requires stacking maybe specify that what drop_nans does is drop samples with nans, or rename argument to reflect that.